libs/core/include/core/random/lfg.hpp

```diff
@@ -79,32 +81,41 @@ class LaggedFibonacciGenerator
     return buffer_[++index_];
   }
 
-  double AsDouble() noexcept
+  /**
+   * @tparam DataType
+   * @return integer uniformly distributed random value 0 or 1
+   */
+  template <typename DataType>
+  meta::IfIsInteger<DataType, DataType> AsFP() noexcept
   {
-    return double(this->operator()()) * inv_double_max_;
+    return static_cast<DataType>(this->operator()() % 2);
   }
 
-  fp128_t AsFP128() noexcept
+  /**
+   * @tparam DataType
+   * @return float or double uniformly distributed random value between 0.0 and 1.0
+   */
+  template <typename DataType>
+  meta::IfIsFloat<DataType, DataType> AsFP() noexcept
   {
-    auto fp128_u_max_ = static_cast<uint64_t>(fp128_t::FP_MAX);
-
-    auto fp_val = static_cast<fp128_t>(this->operator()() % fp128_u_max_);
-    return fp_val / fp128_t::FP_MAX;
+    return static_cast<DataType>(static_cast<double>(this->operator()()) * inv_double_max_);
```

Comment:
Please delete this method and move the implementation directly into AsDouble(). This will avoid 2 casts, and make the intention much clearer for others devs.

Reply:
Okay, so I think the reason you made this function is to allow for a unified interface; in this case, the name AsFP is wrong. You're not returning an FP - you're returning a DataType.
Please rename these Three functions AsType: Then it should be clear when it's called that AsType<Type>() is going to return a Type